iOS mobile share api

theromis opened the pull request:
Adopted https://github.com/Shin-NiL/Godot-Share for godot 4 also implemented same api for Android plugin a bit refactored 
Here is my android implementation as well https://github.com/Lonje/GodotShareData/tree/main/android, inspired by https://github.com/cengiz-pz/godot-android-share-plugin

KoBeWi committed:
Add temporary pivot for rotating multiple 2D nodes

bruvzg committed:
[Windows] Enable crash reporter on MinGW builds, use libbacktrace to generate stack trace from DWARF symbols.

Calinou committed:
Add rotation unit tests for Transform3D

Co-authored-by: Alexander Pruss <alex.pruss@gmail.com>

KoBeWi committed:
Store line change in script navigation history

KoBeWi committed:
Allow loading custom ProjectSettings instance

KoBeWi committed:
Add automatic checking for engine updates

jeronimo-schreyer committed:
Make Advanced Import lights more like the mesh and material editors

Co-authored-by: Aaron Franke <arnfranke@yahoo.com>

smix8 committed:
Add option to bake a mesh from blend shape mix

Adds option to bake a mesh from blend shape mix.

Calinou committed:
Add an editor setting to import resources while unfocused

This can be used to speed up iteration by starting the import process
earlier when saving files in the project folder. This also allows getting
visual feedback on changes without having to click the editor window,
which is useful with multi-monitor setups.

The downside is that this increases idle CPU usage and may steal CPU
time from other applications when importing resources, so this is
disabled by default.

KoBeWi committed:
Add tooltip plugin for AudioStream

KoBeWi committed:
Add a way to invalidate preview cache

KoBeWi committed:
Use existing plugin.cfg

KoBeWi committed:
Add tooltips for tile source types

AThousandShips committed:
[Windows] Improve warning for path casing.

KoBeWi committed:
Don't check resource type assigned in property

KoBeWi committed:
Make varray() variadic

mihe committed:
Add `single`/`double` as export features automatically

jsjtxietian committed:
Update WorkerThreadPool doc to mention every task should be waited
at some point so that any allocated resources can be cleaned up.

KoBeWi committed:
Expose is_part_of_edited_scene()

KoBeWi committed:
Add separate feature tags for editor runtime

Rudolph-B committed:
Fixed particle spread values effecting particle velocity

Koyper committed:
Fix non-embedded Windows resizing to match Viewport content scale factor

KoBeWi committed:
Don't update tree on deselect

ajreckof committed:
Rework MeshInstance3d collision shape creation into a subMenu and add all static body options

Apply suggestions from code review

Co-Authored-By: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

KoBeWi committed:
Automatically add path to built-in shaders

m4gr3d committed:
Fix loading GDExtension dependencies on Android

KoBeWi committed:
Add PropertyListHelper in all simple cases

oxi-dev0 committed:
Fixed unshaded mode lightmaps

MajorMcDoom committed:
Added tokenized search support to Quick Open dialog and FileSystem filter.

AThousandShips committed:
[GDScript] Clean up some unused code

4d49 committed:
Add tab tooltip text

dalexeev committed:
GDScript: Fix continuation lines in `GDScriptTokenizerBuffer`

CrayolaEater committed:
Add Gizmo to display 3D custom AABB

dalexeev committed:
GDScript: Fix object iterator opcodes

Calinou committed:
Add minimum width/height to dialog buttons

This makes buttons with short texts such as "OK" or "Close"
easier to click, especially on a touchscreen.

This is exposed to projects via `buttons_min_width` and `buttons_min_height`
theme items in AcceptDialog (and therefore ConfirmationDialog and FileDialog),
with the default values being 0 to preserve compatibility with existing projects.

bruvzg committed:
[RTL] Move shadow and foreground/background boxes drawing into a separate draw steps.

dalexeev committed:
GDScript: Improve DocGen for non-constant expressions

jsjtxietian committed:
Add progress bar when update scene groups

KoBeWi committed:
Remove stray item icon from tabs menu

AThousandShips committed:
Fix some build errors with `disable_3d=yes`

bruvzg committed:
[iOS Export] Check directory content before deleting old export leftovers.

AThousandShips committed:
Fix import and saving related crashes

* Don't add empty mesh to result when importing obj files
* Check for null resources in `ResourceSaver`

HolonProduction committed:
Completion Tests: Add script to owner

smix8 committed:
Change 2D navigation mesh baking to use floating point coordinates

Replaces internal uses of Clipper2 integer structs to their floating point equivalents.

dalexeev committed:
GDScript: Fix uninitialized local variables not being reset

raulsntos committed:
C#: Make internal properties and property accessors public (but hidden)

Instead of making the accessors `internal` which can break binary compat, make them `public` but hide them with EB never so they don't show up in IntelliSense.

Also, do the same for properties with the `PROPERTY_USAGE_INTERNAL` flag. These properties were not meant to be exposed to scripting, but since they've been public before all we can do now is hide them.

Faless committed:
[MP] Gracefully handle cache confirmation of deleted nodes

It's possible that after sending a cached node reference (e.g. RPC or
static MultiplayerSynchronizer) the reference node is removed from tree
before the remote peer(s) can confirm the referenced path.

To better detect that case, and avoid spamming errors when it happens,
this commit modifies the multiplayer API caching protocol, to send the
received ID instead of the Node path when sending the confirmation
packet.

**This is a breaking change** because it makes the runtime multiplayer
protocol incompatible with previous versions of Godot.

TokageItLab committed:
Add tips detection to bone mapper

Co-authored-by: Lyuma <xn.lyuma@gmail.com>

ajreckof committed:
Fix load subtask not being registered leading to false progress values.

Scarface1809 committed:
Fix error message when removing only child from GraphNode

Fixes #89695.

rburing committed:
Replace Clipper1 library by Clipper2 library

Naros committed:
Expose `goto_help` for GDExtension on `ScriptEditor`

aaronfranke committed:
Add support for explicitly-defined compound triggers in GLTF files

fire committed:
Avoid compiler error with RenderingContextDriverVulkanWindows.

KoBeWi committed:
Add get_process_exit_code() method

Naros committed:
Expose `ClassDB::is_enum_bitfield` as `ClassDB::is_class_enum_bitfield`

bruvzg committed:
﻿[iOS export] Add support for privacy manifest configuration.

bruvzg committed:
[FileAccess] Implement `resize` method.

bruvzg committed:
[macOS export] Detect embedded helper executables using MachO header.

AThousandShips commented:
Please open a [proposal](https://github.com/godotengine/godot-proposals/issues) to track the support and details of this feature

theromis commented:
Converted to draft, asked code owners what do they think, sorry, worked all night to finalize this 2 plugins, took about fiew days to make them work.

AThousandShips commented:
I'm not talking about asking them, I'm talking about referencing their work, you need to cite them for their licenses, so that their work is acknowledged (The first link is under the MIT license which requires this)

AThousandShips commented:
You should copy the license text from their version and add it at the top of the code you copied, after our own header

theromis commented:
added forgotten MIT licenses

AThousandShips commented:
Unrelated to this PR though, was just a suggestion :) It'd still need to be done in the sources here

But first you should write a proposal, best to get that up and discussed before doing any more work

theromis commented:
Sounds good, will do it today's evening

TokageItLab committed:
Separate value/bezier track caching process in AnimationMixer

aaronp64 committed:
Fix theme_override tooltip caching

theme_override properties have some special handling in EditorInspector.  When caching documentation info for properties, the theme_item_name value (used only for theme_overrides) was not being cached, so it would only be included on first use.  This meant that theme_override property tooltips would show up in the editor the first time a node was selected, but not again after selecting something else/reselecting.  Added theme_item_name to caching to fix this.

Fixes #90395

dsnopek committed:
Fix GDExtension hot reload for classes not created via `ClassDB::instantiate()`

theromis commented:
@AThousandShips iOS build failed but locally it builds fine:
```
> scons p=ios target=template_release
scons: Reading SConscript files ...
Auto-detected 16 CPU cores available for build parallelism. Using 15 cores by default. You can override it with the -j argument.
Building for platform "ios", architecture "arm64", target "template_release".
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[  8%] Generating modules/modules_enabled.gen.h ...
...
[100%] scons: done building targets.
[Time elapsed: 00:00:57.427]
```
what am I missing?

theromis commented:
ok, fixed

bruvzg committed:
[RTL] Rewrite `remove_paragraph` code.

bruvzg commented on modules/share/src/godotShareData.h:
```diff
@@ -0,0 +1,53 @@
+/**************************************************************************/
+/*  godotShareData.h                                                      */
```
Filenames should be in snake_case.

theromis replied:
renamed

bruvzg commented on modules/share/register_types.h:
```diff
@@ -0,0 +1,38 @@
+#ifndef SHARE_REGISTER_TYPES_H
+#define SHARE_REGISTER_TYPES_H
+
+#include "modules/register_module_types.h"
```
```suggestion
#include "modules/register_module_types.h"

```

bruvzg commented on modules/share/src/godotShareData.h:
```diff
@@ -0,0 +1,53 @@
+	GodotShareData *instance;
+
+public:
+	void shareText(const String &title, const String &subject, const String &text);
+	void shareImage(const String &path, const String &title, const String &subject, const String &text);
```
```suggestion
	void share_text(const String &p_title, const String &p_subject, const String &p_text);
	void share_image(const String &p_path, const String &p_title, const String &p_subject, const String &p_text);
```

Same for the methods, argument names should have `p_` prefix.

bruvzg commented on modules/share/src/godotShareData.h:
```diff
@@ -0,0 +1,53 @@
+protected:
+	static void _bind_methods();
+
+	GodotShareData *instance;
```
```suggestion
	GodotShareData *singleton;
```

bruvzg commented on modules/share/src/godotShareData.mm:
```diff
@@ -0,0 +1,100 @@
+}
+
+void GodotShareData::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("shareText"), &GodotShareData::shareText);
+	ClassDB::bind_method(D_METHOD("shareImage"), &GodotShareData::shareImage);
```
```suggestion
	ClassDB::bind_method(D_METHOD("share_text", "title", "subject", "text"), &GodotShareData::share_text);
	ClassDB::bind_method(D_METHOD("share_image", "path", "title", "subject", "text"), &GodotShareData::share_image);
```

bruvzg commented on modules/share/src/godotShareData.mm:
```diff
@@ -0,0 +1,100 @@
+	}
+}
+
+void GodotShareData::shareImage(const String &path, const String &title, const String &subject, const String &text) {
```
Not sure if it's useful with `path` as an argument (won't be able to access anything in the PCK, for example). This probably should take `Ref<Image>` instead and convert it to the `UImage` directly.

theromis replied:
Same API for android, internally in kotlin it doing
```
        val f = File(path)
        val uri: Uri = try {
            FileProvider.getUriForFile(activity!!, activity!!.packageName+fileProvider, f)
        } catch (e: IllegalArgumentException) {
            Log.e(logTag, "The selected file can't be shared: $path", e)
            return
        }
```
String-path just easier to cross"provide"

bruvzg commented on modules/share/config.py:
```diff
@@ -0,0 +1,8 @@
+def can_build(env, platform):
+    return env["platform"] in ["ios"]
```
Fully platform dependent module with bound API can't be properly documented, and inconvenient to use from the code. We might want to add some way around it, but since this functionality can be implemented at least on 3 platforms (macOS, iOS, Android), it probably should have always available platform independent base class subclasses for specific platforms.

theromis replied:
for now it available on ios and android, not sure about macos, for android I have plugin, which trying to convert into part of godot core

bruvzg replied:
macOS API - https://developer.apple.com/documentation/appkit/nssharingservicepicker?language=objc

theromis replied:
This can be added as a next step, maybe even windows right now have this kind of api

bruvzg commented on modules/share/src/godotShareData.mm:
```diff
@@ -0,0 +1,100 @@
+#import "app_delegate.h"
+
+GodotShareData::GodotShareData() {
+	ERR_FAIL_COND(instance != NULL);
+	instance = this;
```
```suggestion
	ERR_FAIL_COND(instance != nullptr);
	instance = this;
```

bruvzg commented on modules/share/src/godotShareData.mm:
```diff
@@ -0,0 +1,100 @@
+}
+
+GodotShareData::~GodotShareData() {
+	instance = NULL;
```
```suggestion
	instance = nullptr;
```

AThousandShips commented on modules/share/register_types.cpp:
```diff
@@ -0,0 +1,43 @@
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include <version_generated.gen.h>
```
Not needed

AThousandShips commented on modules/share/register_types.cpp:
```diff
@@ -0,0 +1,43 @@
+#include <version_generated.gen.h>
+
+#include "core/config/engine.h"
+#include "register_types.h"
```
Should be at the top

AThousandShips replied:
Yes, like so:
```cpp
#include "register_types.h"

#include "core/config/engine.h"
```

theromis replied:
already pushed change, thanks

AThousandShips commented on modules/share/src/godot_share_data.h:
```diff
@@ -0,0 +1,53 @@
+#define GODOTSHAREDATA_H
+
+#include "core/object/ref_counted.h"
+#include <version_generated.gen.h>
```
Also not needed

AThousandShips commented on modules/share/src/godot_share_data.h:
```diff
@@ -0,0 +1,53 @@
+protected:
+	static void _bind_methods();
+
+	GodotShareData *singleton;
```
```suggestion
	static GodotShareData *singleton;
```

AThousandShips commented on modules/share/src/godot_share_data.h:
Shouldn't be in a sub directory

AThousandShips commented on modules/share/src/godot_share_data.mm:
Same here

theromis commented:
seems like everything passes except some crazy test killed core

AThousandShips commented on modules/share/godot_share_data.mm:
```diff
@@ -0,0 +1,100 @@
+
+#import "app_delegate.h"
+
+GodotShareData::GodotShareData() {
```
```suggestion
GodotShareData *GodotShareData::singleton = nullptr;

GodotShareData::GodotShareData() {
```

cengiz-pz commented:
Has anyone successfully tested image sharing on iOS?  It may not be working.

theromis commented:
@AThousandShips should this copyright added? from https://github.com/Shin-NiL/Godot-Share
```
--- a/modules/share/godot_share_data.mm
+++ b/modules/share/godot_share_data.mm
@@ -7,6 +7,7 @@
 /**************************************************************************/
 /* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
 /* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/* Copyright (c) 2017                                                     */
 /*                                                                        */
 /* Permission is hereby granted, free of charge, to any person obtaining  */
 /* a copy of this software and associated documentation files (the        */
```

akien-mga commented:
> @AThousandShips should this copyright added? from https://github.com/Shin-NiL/Godot-Share
> 
> ```
> --- a/modules/share/godot_share_data.mm
> +++ b/modules/share/godot_share_data.mm
> @@ -7,6 +7,7 @@
>  /**************************************************************************/
>  /* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
>  /* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
> +/* Copyright (c) 2017                                                     */
>  /*                                                                        */
>  /* Permission is hereby granted, free of charge, to any person obtaining  */
>  /* a copy of this software and associated documentation files (the        */
> ```

There's no name so it's not doing anything.

theromis commented:
sounds like licenses in headers are fine

vnen committed:
GDScript: Resolve types from autoload scenes

When the autoload is a scene, it now extracts the script from it and
use it to further infer nested types.

Calinou committed:
Exit with an error if using `--test` on binary without unit tests compiled

This makes it clearer that `tests=yes` should be used at compile-time.
Previously, the engine would run as if the argument wasn't passed,
which can be confusing.

theromis commented:
merged 2 functions into one, want to set default default values like:
```
    void share(const String &p_text, const String &p_subject = "", const String &p_title = "", const String &p_path = "");
```
but app crashes, what the right way of doing it?

dalexeev committed:
Editor: Display deprecated/experimental messages in tooltips

permelin committed:
Delaunay3D: Improve triangulation

Repiteo committed:
Core: Use `<type_traits>` where applicable

theromis commented:
@AThousandShips what do you think about merging this?

AThousandShips commented:
I can't speak to it as I'm not using iOS or am on the team for that

theromis commented:
@cengiz-pz it works for me, but crashed with default empty values, but it works if I set ""
@akien-mga may be you can help about next steps?

permelin committed:
LightmapGI: Reduce warnings and increase probe accuracy

vnen committed:
GDScript: Assume constructor to be accessible from class

Since the constructor is accessed directly from the class, even if not
marked `static`, it can be retrieved as a Callable without an instance.

theromis commented:
also potential improvement run
```
String path = fix_path(p_path).simplify_path();
```
for path but `fix_path` not exists in java

Rindbee committed:
Make the loading scene open in the current scene tab if the current scene is empty

BMagnu committed:
Fixes the issue where setting global rotation would reset scale.

This was caused because the basis for the target global transform was entirely recreated. This process did not account for the scale of the basis of the current global transform.
This PR amends this by scaling the recreated basis for the global transform by the current global scale of the Node3D.
Note that this scaling has to be done from the current global scale and not from the local scale of the Node3D, otherwise issues would arise if parents of this Node3D would be scaled.

clayjohn committed:
Warn users when assigning VERTEX directly to POSITION due to compatibility breakage from reverse z changes

rune-scape committed:
GDScript: invalidate GDScriptParserRef when reloading

Commit:
Automatically generate the Android debug keystore

Automatically generate the Android debug keystore when the Java SDK path is specified.

Rindbee committed:
Update undo redo allowed after switching scene tab

It has different status in different tabs.

thimenesup committed:
Fix incorrect bounds check in rendering device push constant

smix8 committed:
Warn that navigation mesh baking from Meshes is bad for runtime performance

Warns that navigation mesh baking from Meshes is bad for runtime performance.

YeldhamDev committed:
Fix submenus erroneously closing on some occasions

Malcolmnixon committed:
Rework XR Trackers to have a common ancestor. Allow creation of XRNode3D to drive node positions and visibility.

Muller-Castro committed:
Fix folder colors not present in editor dir dialog

ajreckof committed:
Fix going forward backward in editor selection history.

raulsntos committed:
C#: Fix ExprMatch for case sensitive matching

We were accidentally stopping the recursion when matching in the case sensitive scenario.

Took the opportunity to also rename the private method to follow the Core method naming more closely so it's easier to compare the implementations in the future.

Also, the private method now uses `ReadOnlySpan<char>` to avoid allocating strings on each recursion.

jsjtxietian committed:
Fix editor's file explorer will show directories that should be skipped

Faless committed:
mbedTLS: Fix build without the module on MinGW

The `include/mbedtls/check_config.h` header file has a generic check
which requires `MBEDTLS_PLATFORM_C` to be defined on Windows when
building with MinGW and ancient MSVC versions.

The few mbedTLS files we include in builds without the module don't
really require anything from the mbedTLS platform file beside the
`mbedtls_platform_zeroize` function which we already override with a
custom one by defining `MBEDTLS_PLATFORM_ZEROIZE_ALT`.

So we simply add the same check and force the define in such case.

dalexeev committed:
Core: Fix `RefCounted` handling in `marshalls.cpp`

Faless committed:
[Web] Fix serve.py utility on Windows

IPv6 dual stack is disabled by default, and Windows resolves wildcard
addresses to an IPv6 by default, so connecting through the local IPv4
address would not work.

This enables IPv6 dual stacking for the HTTP server by default like done
in upstream python when launching the module from CLI.

kleonc committed:
Fix TileSet editor create/remove tiles selection rect drawing

Commit:
Fix leakage of JNI object references

Fixes https://github.com/godotengine/godot/issues/87548

bruvzg committed:
Improve TextEdit/LineEdit word selection.

aaronfranke committed:
Move NavigationMeshSourceGeometryData(2D/3D) to the 2D/3D subfolders

tbreese3 committed:
Allow closing scene tab preview using Escape

Calinou committed:
Remove unused Make Streamable option from scene import dialog

timothyqiu committed:
Add project setting for root node auto translate mode

yorickdewid committed:
FIX: Return if TCP is in STATUS_CONNECTED

If the `StreamPeerTCP` is polled and the TCP connection is `STATUS_CONNECTED` it should return after polling netsocket. Without `return` poll keeps calling `_sock->connect_to_host` and `connect()`.

smix8 committed:
Update NavigationRegion test rid function

Updates NavigationRegion test rid function from deprecated get_region_rid() to use newer get_rid().

akien-mga committed:
Update AUTHORS and DONORS list

smix8 committed:
Remove experimental constrain_avoidance from NavigationRegion2D

Removes experimental constrain_avoidance feature from NavigationRegion2D.

aaronfranke committed:
Fix incorrect name of Basis global scale getter

kleonc committed:
Fix transform of Y-sorted branch-root

dalexeev committed:
Core: Fix binary serialization of objects in typed arrays

* Allow typed arrays to be encoded even if `p_full_objects` is `false`.
* Use `Array[EncodedObjectAsID]` instead of the encoded element type
  when decoding an array if `p_allow_objects` is `false`.

groud committed:
Fix TileMap enabling Y-sort on child nodes

vnen committed:
GDScript: Warn when enum variable has no default

The default will always be set to `0`, so if it's not a valid value in
the enum, the warning is shown.

dsnopek committed:
Revert `OpenXRHand` to its pre-`SkeletonModifier3D` state

dsnopek committed:
Mark `OpenXRCompositionLayer` and its children as experimental

Calinou committed:
Increase size of WorldBoundaryShape to be much larger

The size was previously 20,000 pixels or units, which could be easily
reached in many projects. It is now 2,000,000,000,000,000 pixels or units,
which is larger than the supported coordinate space with a single-precision
build, and still very large in a double precision build.

AThousandShips committed:
[Doc] Clarify behavior of `FileAccess.get_line`

Specifies that the line excludes any newline or carriage return
characters.

Naros committed:
Defer call to set_break_language

lyuma committed:
Add SkeletonIK3D get/set_interpolation compat from #87888

Commit:
Only define ENGINE_UPDATE_CHECK_ENABLED in editor builds

bruvzg committed:
[macOS] Fix native file dialog with empty filter list.

KoBeWi committed:
Don't show scene group progress for 1 scene

m4rr5 committed:
Added a specific check for the corner case that went wrong.

Chaosus committed:
Fix incorrect behavior at expanding an output port in visual shader node

lyuma committed:
Fix FBX and glTF when root nodes are skeleton bones

Set p_scene_parent to the skeleton to guarantee BoneAttachment3D nodes are added as a child of the active skeleton.
Use get_owner() to go all the way up when calculating the root node in generate_scene

timothyqiu committed:
Fix potential infinite loop when using global menu

vnen committed:
GDScript: Suppress unused constant warning with underscore

The warning message mentions that local constants prefixed with `_` does
not generate the warning. This commit actually implements this warning
suppression.

ydeltastar committed:
Fix RayCast3D's debug draw transform not updating

Commit:
Fix issue causing the `copyAndRename*` task to fail on occasions on Windows machines

Gradle automatically handles up-to-date checks for output files and directories. This behavior sometimes causes the `copyAndRename*` task to fail on Windows machines when gradle tries to check on existing files in the output directories it doesn't have access to.
To fix the issue, we disable this gradle behavior following the instructions in https://docs.gradle.org/8.2/userguide/incremental_build.html#sec:disable-state-tracking

akien-mga committed:
Tests: Disable NavigationServer3D async bake test with race condition

KoBeWi committed:
Disable Create button for abstract classes

RandomShaper committed:
WorkerThreadPool: Polish yielding (fix corner case, remove misleading warning)

Arnklit committed:
fix: avoid shader crash on null vector and negative x vector

Repiteo committed:
SCons: Convert `env_base` to `env`

KoBeWi committed:
Avoid double editing when clicking AnimatedSprite

kitbdev committed:
Don't edit current when changing docks

KoBeWi committed:
Reuse single erase pattern in tile editor

clayjohn committed:
Use a src rect for copying from screen with CanvasGroup in the mobile backend

Geometror committed:
[Compatibility] Add stub for VisualShaderNodeComment

timothyqiu committed:
Fix folder color not cleared for removed subfolders

clayjohn committed:
Revert change to default depth clear value in draw_list_begin

Calinou committed:
Strip ANSI escape codes from file logging

Text editors cannot display ANSI escape codes, so these should be
stripped from log files to ensure readability.

Since this uses a regex, this works both for `print_rich()`
and manually inserted ANSI escape codes.

Co-authored-by: bruvzg <7645683+bruvzg@users.noreply.github.com>

lyuma committed:
Expose create_from_arrays in SurfaceTool and cleanup some naming

bruvzg committed:
Fix input map shortcuts incorrectly edited on cancel.

timothyqiu committed:
Fix TreeItem button handling

- Fix incorrect tooltip and `get_button_id_at_position()` when column
  title is visible and when RTL layout is used
- Take `button_margin`, `h_separation`, and `item_margin` into account

paulloz committed:
Change order of operation for C# types reloading

We now deserialize callables before reloading property states, in case a property is doing anything with the callable in its getter and/or setter.

radiantgurl committed:
Implement remove_user_signal()

Co-authored-by: Timothe Bonhoure <tbonhoure@ymail.Com>
Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

bruvzg committed:
[Button] Fix theme elements size rounding errors not keeping sufficient space for the text.

timothyqiu committed:
Revert "Correctly display tooltips for buttons in Tree when they overlap cell content"

This reverts commit 8278a5f274eda7484b4f31a5acb4af34b9d4fc39.

KoBeWi committed:
Fix folder scan replacing project list

BlueCube3310 committed:
LightmapGI: Fix seam blending with directional lightmaps

timothyqiu committed:
Fix buttons offset to the right when dialog is at minsize

Repiteo committed:
SCons: Utilize native Variable alias functionality

Repiteo committed:
SCons: Disable `show_progress` with ninja

vnen committed:
GDScript: Implement get_dependencies()

The parser and analyzer now track the dependencies of the script and
return the list when the resource loader ask for them.

What is considered a dependency:

- Any `preload()` call.
- The base script this one extends.
- Any identifier, including types, that refers to global scripts.
- Any autoload singleton reference.

aaronp64 committed:
Check for NaN in set_volume_db functions

Added check in AudioStreamPlayer, AudioStreamPlayer2D, and AudioStreamPlayer3D set_volume_db functions to prevent setting volume to NaN, and give an error.  Using NaN for volume and playing the AudioStreamPlayer could prevent all audio from playing, even from other AudioStreamPlayers.

Fixes #88133

radiantgurl committed:
Remove experimental flag.

RandomShaper committed:
WorkerThreadPool: Fix yield-over for not-yet-started tasks

Repiteo committed:
Core: Integrate initalizer lists for Variant

KoBeWi committed:
Don't translate ColorPicker color labels

Zylann committed:
Fix crash when GraphFrame sb_to_draw_panel is not a StyleBoxFlat

smix8 committed:
Add navigation source geometry parser callbacks

Adds navigation source geometry parser callbacks so that externals can hook their own geometry into the navigation mesh baking process.

KoBeWi committed:
Use minor version in EditorSettings file name

cosparks committed:
Fixes SDF Collision Enable/Disable
- sets LightOccluderInstance field when sdf collision is updated
- adds check for light occluder sdf_collision field in 2d renderers

timothyqiu committed:
Fix offset of node filter menu separator icon

timothyqiu committed:
Use compatible text resource format when possible

lyuma committed:
Set animation step from importers. Increase default step from 10 to 30 FPS.

m4gr3d committed:
Collapse the gdextension arguments into the `GDExtensionData` struct

This is used to reduce the number of arguments to `OS::open_dynamic_library(...)`.

KoBeWi committed:
Don't store TileMapLayer data if empty

clayjohn committed:
Avoid writing out of range data over valid data in CVTT decompress methods when decompressing small mip levels

clayjohn committed:
Ensure minimum block size is respected when doing GPU to CPU copies of compressed textures

rburing committed:
Fix move_and_slide wall slide acceleration (3D)

When travel is high enough, keep the global position resulting from the
move_and_collide call, and set the motion to the remainder from the
move_and_collide call. This ensures travel is taken into account once,
rather than twice.

Naros committed:
Expose ClassDB::class_get_property_default_value method

clayjohn committed:
Exit light calculation early when pixel outside of light bounding rectangle

This hugely improves the performance of rendering PointLight2Ds

theromis commented:
@AThousandShips is there any examples of android java plugin in core modules?

aaronfranke committed:
Rename internal is_ascii_char to is_ascii_alphabet_char

smix8 committed:
Add NavigationMeshSourceGeometryData append functions

Adds append functions to NavigationMeshSourceGeometryData.

AThousandShips committed:
[Web] Fix building for web on Windows

On Windows the command for emscripten are provided as `.bat` files,
which causes the compiler version check to fail without `shell=True`

timothyqiu committed:
Fix TabContainer not redrawing after toggling tab icon

AThousandShips commented:
Not my area so I can't answer

rhofour committed:
Have GDScriptWorkspace::list_script_files respect .gdignore.

theromis commented:
@AThousandShips who to ask?

AThousandShips commented:
Just wait for someone on the relevant teams to come around, the relevant teams have been requested, or ask in the [chat](https://chat.godotengine.org/)

theromis commented:
Thank you, waiting

Lateasusual committed:
GDExtension: Fix missing library path breaking hot reloading

TokageItLab committed:
Correct AnimationNode document where it mentions multiple inputs

Alex-Mayo committed:
Clarify docs for ParticleProcessMaterial emission_box_extents

raulsntos committed:
C#: Don't return MethodInfo for overloaded methods

This means the GDScript analyzer loses the method signature information so it can't do type checking for the parameters.

aaronfranke committed:
Organize existing code for editor plugins

Fulanko committed:
Fix incorrect c# in OS.get_cmdline_args

TheSofox committed:
Enabled 'Scrolling' signal when scrolling with middle mouse on RichTextLabel/ScrollContainer

aaronp64 committed:
Handle compressed images in TileSetAtlasSource::_create_padded_image_texture

Check for compressed image and decompress if needed

Fixes #78755

theromis committed:
Add MANAGE_GAME_MODE option for Firebase Analytics

m4gr3d committed:
Cleanup Android file errors

Follow up to https://github.com/godotengine/godot/pull/90403

killzebug committed:
Suppressed expected errors for uncached resources.

KoBeWi committed:
Hide update status label when no action is required

Malcolmnixon committed:
Subscribe to correct tracker signals

bqqbarbhg committed:
Resolve bind poses from FBX clusters instead of FBX poses

Turns out that the information in FBX Pose objects is relatively often broken.
Using skin cluster bind poses seems more reliable, but cannot capture the bind pose of the root bone.

timothyqiu committed:
Fix TabContainer not updating content rect after changing tab title

timothyqiu committed:
Fix wrong Tree column title position

lyuma committed:
fbx: Avoid name conflict with humanoid "Root" bone

The importer forces name uniqueness, even for the root. "RootNode" is less likely to conflict.

bruvzg committed:
[iOS] Fix sensors orientation.

BastiaanOlij committed:
Fix issue with copy shader not working in multiview

bruvzg committed:
[DBus] Process file dialog callback in the main event loop instead of using deferred call.

Repiteo committed:
SCons: Only use alias if SCons v4.3 or later

akien-mga committed:
Revert "Don't edit current when changing docks"

This reverts commit 55711b2ee849163135640cda0f00fcdcb0819af5.
Fixes #91046.

KoBeWi committed:
Remove uv_editor_bounds editor setting

AThousandShips committed:
[CI] [macOS] Fix CI with new GitHub runners

New `macos-latest` run on an arm64 image instead of x86_64 breaking the
editor workflow.

Bn-Fang committed:
Fixed Typo in Camera 2D node's anchor mode

Fixed the Camera 2D node's anchor mode descriptor from "Fixed TopLeft" to "Fixed Top Left"

aaronp64 committed:
Fix errors/crashes related to skipped imports

- Added check for "animation/fps" key before attempting to use it in EditorSceneFormatImporterBlend::import_scene, to give error instead of crashing

- Don't show "Advanced..." button if last import used "Keep File" or "Skip File"

- Don't try to call ResourceLoader::load on kept/skipped file when changing importer, which would give an error

Fixes #90324

paulloz committed:
Enforce using .NET SDK >= 8 in modules/mono/

Repiteo committed:
SCons: `scons_version` to environment variable

KoBeWi committed:
Further speed up closing multiple scripts

Commit:
Fix errors and warnings when loading Skeleton2D Modifications

Fixes #73247

timothyqiu committed:
Fix error when moving FileSystem dock

timothyqiu committed:
Fix Tree minimum size calculation

- Take scroll bar space and column title buttons into account
- Fix first column min size sometimes missing one level of indent
- Fix cell min size ignoring text overrun behavior and item inner margin
- Update min size when `hide_root` or `column_title_visible` changes

Wrong description of `item_inner_margin_*` constants is also fixed

BastiaanOlij committed:
Fix issue in shadow to opacity

Commit:
Reset state of SkeletonModification2D when loaded from cache

timothyqiu committed:
Fix syntax error in camera_attributes_set_exposure documentation

bruvzg committed:
Add symlink API support for Windows, expose symlink methods.

nagidev committed:
Fixed port cache calculation in GraphNode

timothyqiu committed:
Add text to icon-only interpolation OptionButton

smix8 committed:
NavigationRegion test suppress warning

Suppresses warning about source geometry parsing from visual meshes in NavigationRegion test.

AThousandShips committed:
[CI] [macOS] Build for both `arm64` and `x86_64`

Creates a universal build, to catch discrepancies on different
architectures

akien-mga committed:
Sync controller mappings DB with SDL 2 community repo

Synced with mdqinc/SDL_GameControllerDB@5b4efa3a2015552cec4cb809169d29615202e9f2

kleonc committed:
Fix Path3D tilt gizmo raycasting against local plane

KoBeWi committed:
Clarify transform inheritance of CanvasItem nodes

KoBeWi committed:
Register text enter for script base class

deralmas committed:
Wayland: Improve compositor compatibility by allowing older globals

Previously we pretty much hardcoded most of the globals we requested,
causing compatibility issues with certain compositors like Weston, which
support only some pretty old versions or miss some more advanced
protocols.

To put fuel on the fire, we also errored out when certain protocols
weren't available, despite us being able to boot a game just fine (but
obviously with a degraded featureset).

The solution is to simply allow all the way from version 1 to the
current latest, adding some compatibility code (such as for older
`wl_output`s or newer `wl_pointer`s).

While we're at it, this commit also fixes a few typos and naming inconsistencies
I found.

Koyper committed:
[PopupMenu] Fix incorrect vertical scroll bar visible at fractional content scale

Konstantin-Kretov committed:
Fill copyright field in .csproj files

This ensures that nuget packages will have both license and copyright fields filled.

dalexeev committed:
GDScript: Fix test `reset_uninit_local_vars.gd` failure

dsnopek committed:
Fix GDExtension documentation disappearing after hot-reload

Malcolmnixon committed:
Remove the Hand enum from XRHandTracker to fix name-collision with the hand property of the base class.

Co-Authored-By: David Snopek <191561+dsnopek@users.noreply.github.com>

basicer committed:
Add LIGHT_VERTEX to fragment shader

Adds a new variable to the fragment shader to specify
the vertex position used when calculating lighting.

ckaiser committed:
Use TextEdit caret color for minimap highlight

kus04e4ek committed:
Don't ignore `audio/driver/output_latency.web`

bruvzg committed:
[Windows] Improve console IO redirection.

dsnopek committed:
Fix ptrcalls to static GDExtension methods

Arnklit committed:
Add show in filesystem option for file favorites

Calinou committed:
Allow compressed texture array and cubemap types in shader globals editor

DarioSamo committed:
Fix crash on compatibility fallback when vkCreateRenderPass2KHR is unavailable.

clayjohn committed:
Add adjustments and color correction to Compatibility renderer

markusstephanides committed:
Add no-change check to Label3D

Bromeon committed:
GDExtension: provide `free_property_list_func` with length of array

KoBeWi committed:
Misc code cleanup

emanvidmaker committed:
FileAccess.xml had conflicting function names in its example code

In the example code the declared load() function conflicts with global scope load(). so if you copy pasted it in godot 4.2.2 you would get a "Too few arguments for "load()" call. Expected at least 1 but received 0." error since it doesn't override global scope load().

kitbdev committed:
Don't use vscroll when fit to content height in TextEdit

clayjohn committed:
Properly calculate penumbra for soft shadows with reverse z

Also fix a related bug where the DirectionalLight3D size was ignored unless a positional light with soft shadows touched the mesh

timothyqiu committed:
Allow setting icon max width for TabContainer

jsjtxietian committed:
Fix `SurfaceTool.create_from_arrays` crash with bad index

Faless committed:
[Web] Fix closure compiler typedef annotation

The typedef annotation is expected to come bofre a var (or const) since
it's most commonly used in externs. Use an inline definition instead.

akien-mga committed:
Fix GCC 14 -Wtemplate-id-cdtor warnings

Fixes #91206.

magian1127 committed:
Implement tooltips for shader uniform in the inspector.

using regular expressions

Faless committed:
[Core] Use unztell64 in FileAccessZIP to ensure 64 bit return

DeeJayLSP committed:
ResourceImporterWAV: remove unnecessary var declaration

Lateasusual committed:
Fix inverted GPUParticlesCollisionHeightField3D

Repiteo committed:
SCons: Colorize warnings/errors during generation

brennennen committed:
doc fix typo that flipped the description of
window_set_max_size and window_set_min_size

TokageItLab committed:
Make adding track process to set interpolation type on creating RESET

TokageItLab committed:
Make activating on ready in AnimationPlayer respect the property value

AThousandShips committed:
Fix unsafe uses of `Callable.is_null()`

`Callable.is_null()` is not equivalent to `!Callable.is_valid()` and
doesn't guarantee the call is valid.

timothyqiu committed:
Only add warning label for Layout group in Control category

ajreckof committed:
Fix RichTextLabel "Scroll Following" randomly stops working.

akien-mga committed:
Revert "glad: Re-generate files with glad 2.0.6"

This reverts commit 0f96251afe19007ff8965b00f00efc1d9b1d6ea1.

Fixes #90879.

theromis commented:
now I broke PR, let me recreate it

AThousandShips commented:
In the future it's better to try and solve the issue, asking for help, as this loses a lot of feedback and discussions, this would be easily solved with:
```
git reset --hard 066be2c
```
And then redoing the failed rebase

theromis commented:
Thank you, will try this approach next time
